spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -177,15 +254,25 @@ def __init__(self, parent, id_,
         # --- Dialog manager
         self.dialog_manager = DialogManager()
 
-        # Poll for stderr changes
-        self.stderr_mtime = 0
-        self.stderr_timer = QTimer(self)
-        self.stderr_timer.timeout.connect(self.poll_stderr_file_change)
-        self.stderr_timer.setInterval(1000)
-        self.stderr_timer.start()
+        self.stderr_obj = None
+        self.stdout_obj = None
+        self.fault_obj = None
+        self.std_poll_timer = None
+        if not self.is_external_kernel:
+            # Can not connect for external kernels
+            self.stderr_obj = Std_File(self.std_filename('.stderr'))
+            self.stdout_obj = Std_File(self.std_filename('.stdout'))
+            self.std_poll_timer = QTimer(self)
+            self.std_poll_timer.timeout.connect(self.poll_std_file_change)
+            self.std_poll_timer.setInterval(1000)
+            self.std_poll_timer.start()
+            self.shellwidget.executed.connect(self.poll_std_file_change)
+        if self.hostname is None:
+            # Can not read file that is not on this computer
+            self.fault_obj = Std_File(self.std_filename('.fault'))
 
     def __del__(self):
-        """Close threads to avoid segfault"""
+        """Close threads to avoid segfault."""

```

Comment:
This is likely an issue as well, but not covered in this PR. Don't close spyder while restarting a kernel?